src/OneWayBondingCurve.sol

```diff
@@ -60,23 +55,32 @@ contract OneWayBondingCurve {
 
     /// @notice Purchase USDC for BAL
     /// @param amountIn Amount of BAL input
+    /// @param withdrawFromAave Whether to receive as USDC (true) or aUSDC (false)
     /// @return amountOut Amount of USDC received
     /// @dev Purchaser has to approve BAL transfer before calling this function
-    function purchase(uint256 amountIn) external returns (uint256) {
+    function purchase(uint256 amountIn, bool withdrawFromAave) external returns (uint256) {
```

Comment:
might make sense to rename `withdrawFromAave` to `toUnderlying` to be in line with aave contracts param naming.
No strong opinion though.